### ISSUE TITLE
Bug: require-await false negative on empty function body

### ISSUE THREAD

**Issue by kirkwaiblinger** [eslint/eslint#18338]
### Environment

Node version: 
npm version: 
Local ESLint version: 
Global ESLint version: 
Operating System: 

### What parser are you using?

Default (Espree)

### What did you do?

```js
const a = async () => {};
```

### What did you expect to happen?

error on line 1: "Async arrow function has no 'await' expression"

### What actually happened?

no error

### Link to Minimal Reproducible Example

https://eslint.org/play/#eyJ0ZXh0IjoiY29uc3QgYSA9IGFzeW5jICgpID0+IHt9OyIsIm9wdGlvbnMiOnsicnVsZXMiOnsicmVxdWlyZS1hd2FpdCI6WyJlcnJvciJdfSwibGFuZ3VhZ2VPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjoibGF0ZXN0Iiwic291cmNlVHlwZSI6Im1vZHVsZSIsInBhcnNlck9wdGlvbnMiOnsiZWNtYUZlYXR1cmVzIjp7fX19fX0=

### Participation

- [X] I am willing to submit a pull request for this issue.

### Additional comments

[The doc page](https://eslint.org/docs/latest/rules/require-await#rule-details) says "This rule warns async functions which have no `await` expression." This condition is plainly met.

**Comment by ota-meshi** [comment]
Hi @kirkwaiblinger.

The documentation says that `Allow empty functions.`. I think that's intentional.
<img width="757" alt="image" src="https://github.com/eslint/eslint/assets/16508807/1623c3cb-f48b-42cf-a575-e0af9173a0ca">
https://eslint.org/docs/latest/rules/require-await#rule-details

You can also use the no-empty-function rule.
https://eslint.org/docs/latest/rules/no-empty-function

**Comment by kirkwaiblinger** [comment]
@ota-meshi 
Oh, sorry, I totally missed that in the examples. Closing :)
